src/dotnet/Chat.Service/ChatAuthorsBackend.cs

```diff
@@ -189,6 +189,45 @@ public virtual async Task<ChatAuthor> Create(IChatAuthorsBackend.CreateCommand c
         return chatAuthor;
     }
 
+    public virtual async Task<ChatAuthor> ChangeHasLeft(IChatAuthorsBackend.ChangeHasLeftCommand command, CancellationToken cancellationToken)
+    {
+        var context = CommandContext.GetCurrent();
+        var (authorId, hasLeft) = command;
+        if (Computed.IsInvalidating()) {
+            var invChatAuthor = context.Operation().Items.Get<ChatAuthor>()!;
+            var userId = (string)invChatAuthor.UserId;
+            var chatId = (string)invChatAuthor.ChatId;
+            if (!userId.IsNullOrEmpty()) {
+                _ = GetByUserId(chatId, userId, true, default);
+                _ = GetByUserId(chatId, userId, false, default);
+                _ = ListUserIds(chatId, default);
+                _ = ListUserChatIds(userId, default);
+            }
+            _ = Get(invChatAuthor.ChatId, invChatAuthor.Id, false, default);
+            _ = Get(invChatAuthor.ChatId, invChatAuthor.Id, true, default);
+            _ = ListAuthorIds(chatId, default);
+
+            return default!;
+        }
+
+        var dbContext = await CreateCommandDbContext(cancellationToken).ConfigureAwait(false);
+        await using var __ = dbContext.ConfigureAwait(false);
+
+        var dbChatAuthor = await dbContext.ChatAuthors
+            .SingleOrDefaultAsync(a => a.Id == authorId, cancellationToken)
```

Comment:
do we really need `SingleOrDefaultAsync` here?
https://docs.microsoft.com/en-us/aspnet/core/data/ef-rp/crud?view=aspnetcore-6.0#ways-to-read-one-entity